evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,436 @@
+
+                    this.active = element;
+
+                    if(this.active.length) {
+                        this.active.removeClass("to-left to-right");
+                        this.active.addClass("to-" + direction);
+                        this.active.position();
+                        this.active.addClass("active");
+                    }
+                }
+            };
+
+            function QuickSlider() {
+                this.setup.apply(this, arguments);
+            }
+
+            QuickSlider.prototype = {
+                actions: {
+                    publish: "btn-success",
+                    make_private: "btn-dark",
+                    hide: "btn-danger"
+                },
+
+                buttonTimeout: null,
+
+                setup: function(slider) {
+                    this.slider = slider;
+
+                    for(let action in this.actions) {
+                        let btn = this.slider.find("[data-action=" + action + "]");
+                        btn.click($.proxy(this.press, this, action, true));
+                    }
+
+                    for(let repeat of ["unreviewed", "all"]) {
+                        let btn = this.slider.find("[data-action=repeat-" + repeat + "]");
+                        btn.click($.proxy(this.repeat, this, repeat));
+                    }
+
+                    for(let direction of ["left", "right"]) {
+                        let btn = this.slider.find("[data-action=slide-" + direction + "]");
+                        btn.click($.proxy(this.slide, this, direction));
+                    }
+
+                    let questionnaires = new Layer(slider.find("[data-layer=questionnaire]"));
+                    let questions = new Layer(slider.find("[data-layer=question]"), questionnaires);
+                    this.answers = new Layer(slider.find("[data-layer=answer]"), questions);
+                    this.alert = this.slider.find(".alert");
+                    this.alert.on("transitionend", function() {
+                        $(this).removeClass("to-left to-right");
+                    });
+
+                    this.repeat("start");
+
+                    $(document).keydown($.proxy(this.onKeyDown, this));
+                    this.slider.on("touchstart", $.proxy(this.onTouchStart, this));
+                    this.slider.on("touchmove", $.proxy(this.onTouchMove, this));
+                    this.slider.on("touchend", $.proxy(this.resetTouchSliding, this));
+                    this.updateState();
+                },
+
+                get active() {
+                    if(this.index === this.items.length) {
+                        return this.alert;
+                    }
+
+                    return this.items[this.index];
+                },
+
+                get numReviewed() {
+                    return $.grep(this.items.slice(0, this.index), item => item.is("[data-review]")).length;
+                },
+
+                get numAlreadyReviewed() {
```

Comment:
i was completely confused by the names of these two functions (there are some variables somewhere else as well). i would suggest `numReviewedBefore` and `numReviewedAfter`, but I'm not really happy with that myself. maybe there's something better...

Reply:
Agreed, I'm still collecting ideas to refactor this code and this is definitely on my list